framework/operator.go

```diff
@@ -0,0 +1,205 @@
+package framework
+
+import (
+	"github.com/cenk/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+// Config represents the configuration used to create a new operator framework.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new operator
+// framework by best effort.
+func DefaultConfig() Config {
+	var err error
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	return Config{
+		// Dependencies.
+		Logger: newLogger,
+	}
+}
+
+// New creates a new configured operator framework.
+func New(config Config) (*Framework, error) {
```

Comment:
The framework is now a service like authority which is configurable. The diff is quite bad. The base functionality did not change. See more information below.